core/src/test/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreTest.java

```diff
@@ -463,53 +463,6 @@ public void testKafkaStoreMessageHandlerSameIdSameSchema() throws Exception {
     assertEquals(2, size);
   }
 
-  @Test
-  public void testReplaceDeletedWithNonDeleted() throws Exception {
-    InMemoryCache<SchemaKey, SchemaValue> inMemoryStore = new InMemoryCache<>();
-
-    int id = 100;
-    SchemaKey schemaKey = new SchemaKey("subject", 1);
-    SchemaValue schemaValue =
-        new SchemaValue("subject", 1, id, "schemaString", false);
-
-    SchemaKey schemaKey2 = new SchemaKey("subject2", 1);
-    SchemaValue schemaValue2 =
-        new SchemaValue("subject2", 1, id, "schemaString", false);
-
-    inMemoryStore.put(schemaKey, schemaValue);
-    inMemoryStore.schemaRegistered(schemaKey, schemaValue);
-
-    inMemoryStore.put(schemaKey2, schemaValue2);
-    inMemoryStore.schemaRegistered(schemaKey2, schemaValue2);
-
-    schemaValue2.setDeleted(true);
-    inMemoryStore.schemaDeleted(schemaKey2, schemaValue2);
-
-    assertTrue(inMemoryStore.get(inMemoryStore.schemaKeyById(id)).isDeleted());
-
-    inMemoryStore.replaceMatchingDeletedWithNonDeletedOrRemove(s -> s.equals("subject2"));
-
-    SchemaValue newValue = inMemoryStore.get(inMemoryStore.schemaKeyById(id));
-    assertEquals("subject", newValue.getSubject());
-    assertFalse(newValue.isDeleted());
-  }
-
-  @Test
-  public void testReplaceDeletedWithNonDeletedAfterCompaction() throws Exception {
-    InMemoryCache<SchemaKey, SchemaValue> inMemoryStore = new InMemoryCache<>();
-
-    int id = 100;
-    SchemaKey schemaKey = new SchemaKey("subject", 1);
-    SchemaValue schemaValue =
-        new SchemaValue("subject", 1, id, "schemaString", true);
-
-    // After a compaction, the schema will not be registered but only deleted
-    inMemoryStore.put(schemaKey, schemaValue);
-    inMemoryStore.schemaDeleted(schemaKey, schemaValue);
-
-    inMemoryStore.replaceMatchingDeletedWithNonDeletedOrRemove(s -> s.equals("subject"));
-  }
-
```

Comment:
These tests were for the complicated logic to maintain `(guid -> schemaKey)` and so are no longer needed.